### ISSUE TITLE
Add multi-layer vertical coupling matrix (A) and layer↔mode transforms (Cl2m / Cm2l)

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/finitevolX#67]
## Description

The **A matrix** (vertical coupling from layer thicknesses `H` and reduced gravities `g'`) and its eigendecomposition into **Rossby deformation modes** are core building blocks for any multi-layer QG or layered ocean model. They are used to:

- Decouple the multi-layer Helmholtz equation into independent 2D modal solves (PV inversion)
- Compute Rossby deformation radii from eigenvalues
- Transform fields between layer space and mode space (`Cl2m` / `Cm2l`)

This functionality is present in two reference repos but **completely absent** from finitevolX.

### Reference Implementations

**From [`louity/MQGeometry/qgm.py#L96-L119`](https://github.com/louity/MQGeometry/blob/1511b1a9bbc4b1213017f59305e2ce466b308169/qgm.py#L96-L119):**
```python
def compute_auxillary_matrices(self):
    # A operator
    H, g_prime = self.H.squeeze(), self.g_prime.squeeze()
    self.A = torch.zeros((self.nl,self.nl), **self.arr_kwargs)
    if self.nl == 1:
        self.A[0,0] = 1./(H*g_prime)
    else:
        self.A[0,0] = 1./(H[0]*g_prime[0]) + 1./(H[0]*g_prime[1])
        self.A[0,1] = -1./(H[0]*g_prime[1])
        for i in range(1, self.nl-1):
            self.A[i,i-1] = -1./(H[i]*g_prime[i])
            self.A[i,i] = 1./H[i]*(1/g_prime[i+1] + 1/g_prime[i])
            self.A[i,i+1] = -1./(H[i]*g_prime[i+1])
        self.A[-1,-1] = 1./(H[self.nl-1]*g_prime[self.nl-1])
        self.A[-1,-2] = -1./(H[self.nl-1]*g_prime[self.nl-1])

    # layer-to-mode and mode-to-layer matrices
    ev_A, P = torch.linalg.eig(self.A)
    self.lambda_sq = self.f0**2 * ev_A.real.reshape((1, self.nl, 1, 1))
    self.Cl2m = torch.linalg.inv(P.real)
    self.Cm2l = P.real
    self.rossby_radii = self.lambda_sq.squeeze().pow(-0.5)
```

**From [`louity/qgsw-pytorch/src/qg.py#L35-L58`](https://github.com/louity/qgsw-pytorch/blob/673c8e60288baaf1e426dfc3ddd4266380f781cf/src/qg.py#L35-L58):**
```python
def compute_auxillary_matrices(self):
    # A operator (same tridiagonal construction)
    ...
    # layer-to-mode and mode-to-layer matrices (biorthogonal variant)
    lambd_r, R = torch.linalg.eig(self.A)
    lambd_l, L = torch.linalg.eig(self.A.T)
    self.lambd = lambd_r.real.reshape((1, self.nl, 1, 1))
    R, L = R.real, L.real
    self.Cl2m = torch.diag(1./torch.diag(L.T @ R)) @ L.T
    self.Cm2l = R
```

Note the two repos use slightly different normalization: MQGeometry uses `inv(P)`, while qgsw-pytorch uses the biorthogonal `L^T / (L^T R)` form which is more numerically stable.

## Proposed API

```python
def build_coupling_matrix(
    H: Float[Array, "nl"],        # layer thicknesses
    g_prime: Float[Array, "nl"],   # reduced gravities between layers
) -> Float[Array, "nl nl"]:
    """Build the tridiagonal vertical coupling matrix A.

    A[i,j] encodes the vertical structure coupling between layers
    based on layer thicknesses and reduced gravities.
    """

def decompose_vertical_modes(
    A: Float[Array, "nl nl"],
    f0: float,
) -> Tuple[Float[Array, "nl"], Float[Array, "nl nl"], Float[Array, "nl nl"]]:
    """Eigendecompose A to get Rossby deformation radii and transforms.

    Returns
    -------
    rossby_radii : Array of shape (nl,)
        Rossby deformation radii for each mode.
    Cl2m : Array of shape (nl, nl)
        Layer-to-mode transform matrix.
    Cm2l : Array of shape (nl, nl)
        Mode-to-layer transform matrix.
    """

def layer_to_mode(field: Array, Cl2m: Array) -> Array:
    """Transform a field from layer space to mode space."""

def mode_to_layer(field: Array, Cm2l: Array) -> Array:
    """Transform a field from mode space to layer space."""
```

## Acceptance Criteria

- [ ] `build_coupling_matrix(H, g_prime)` for 1-layer and N-layer cases
- [ ] `decompose_vertical_modes(A, f0)` returning radii, Cl2m, Cm2l
- [ ] `layer_to_mode` and `mode_to_layer` transform functions
- [ ] Unit tests: 1-layer, 2-layer, 3-layer cases with known analytic results
- [ ] Test round-trip: `mode_to_layer(layer_to_mode(field)) ≈ field`
- [ ] Verify computed Rossby radii against known values (e.g., standard ocean params)
- [ ] JAX-compatible (pure functions, jittable)
- [ ] Docstrings with usage examples and physical interpretation

## Priority
**High** — essential for multi-layer QG PV inversion and modal decomposition in SWM.

---

_References:_
- _[louity/MQGeometry/qgm.py](https://github.com/louity/MQGeometry/blob/1511b1a9bbc4b1213017f59305e2ce466b308169/qgm.py#L96-L119)_
- _[louity/qgsw-pytorch/src/qg.py](https://github.com/louity/qgsw-pytorch/blob/673c8e60288baaf1e426dfc3ddd4266380f781cf/src/qg.py#L35-L58)_
